### PR TITLE
Fix #67 last digit zero issue.

### LINKED ISSUE #67: UI issue when estimating the value between pairs
In the exchange page enter the following values:

DGD: 1.2341110

the estimate value in ETH will be:
ETH: 0.000000000000

If you change the last digit to:

DGD: 1.2341111

the estimated value in ETH will be:
ETH: 0.673446601147

It seems that if the last digit is 0, the estimated value will be 0.

### PR DESCRIPTION
**Pull request by whyka**
I did a small change to fix #67.
Anyway, current data flow is not quite clear and I'm doing a refactor.

### CODE REVIEW

**Comment by tranvictor** [comment]
@whyka can you describe a bit more on how you would refactor it?

File: src/js/components/ExchangeForm/TokenSource.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ export default class TokenSource extends React.Component {
             <option key={constants.ETHER_ADDRESS} value={constants.ETHER_ADDRESS}>ETH</option>
             {tokenOptions}
           </select>
-          <input name='token_source' onKeyPress={this.props.onKeyPress} value={toT(this.props.specifiedAmount)} type="number" min="0" step="any" placeholder="Amount to exchange" onChange={this.specifyAmount.bind(this)}/>
+          <input name='token_source' onKeyPress={this.props.onKeyPress} type="number" min="0" step="any" placeholder="Amount to exchange" onChange={this.specifyAmount.bind(this)}/>
```
Comment (tranvictor) [inline src/js/components/ExchangeForm/TokenSource.js]: we also don't need to implicitly `bind(this)` if we use this syntax:
```
specifyAmount = () => {
}
```

**Comment by whyka** [comment]
I know it's syntactic sugar but not a necessarily to fix the issue. It's
better if push another refactor commit separately.

**Comment by whyka** [comment]
Some components are duplicated such as TokenSource and TokenDest. Actually
they should be just Presentational and lift up state to parent Container as
in
https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0?source=linkShare-59f0fe8e86b7-1503278767
.
Many converters and formats string in many places are not good also.

File: src/js/components/ExchangeForm/TokenSource.js (hunk 1 above)
Comment (whyka) [inline src/js/components/ExchangeForm/TokenSource.js]: I know it's syntactic sugar but not a necessarily to fix the issue. It's
better if push another refactor commit separately.

**Comment by tranvictor** [comment]
Ok sounds good.

**Comment by tranvictor** [comment]
also fix #62 #64

**Pull request merged**